spec.bs

```diff
@@ -1707,6 +1707,25 @@ Note: When trusted scoring signals fetches are not batched, |renderURLs|'s [=lis
 
 </div>
 
+<div algorithm>
+To <dfn>send report</dfn> given a [=URL=] |url|:
+
+  1. Let |request| be a new [=request=] with the following properties:
+    :   [=request/URL=]
+    ::  |url|
+    :   [=request/client=]
+    ::  `null`
+    :   [=request/mode=]
+    ::  "`no-cors`"
+    :   [=request/referrer=]
+    :: "`no-referrer`"

```

Comment:
[The implementation] (https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:services/network/public/cpp/resource_request.h;drc=a5150e1903bde84e008ac27d8881d9ccf72e087e;l=147) uses the default 'NEVER_CLEAR'.
What does it map to in [fetch's spec](https://fetch.spec.whatwg.org/#concept-request-referrer)?

Reply:
These aren't the same concepts. In spec you're pointing to https://fetch.spec.whatwg.org/#concept-request-referrer but in the implementation you're pointing to https://fetch.spec.whatwg.org/#concept-request-referrer-policy.

Reply:
So I'd need to see the actual place in the implementation where this request is being set up.

Reply:
oh right, sorry about that. It seems ["unsafe-url"](https://w3c.github.io/webappsec-referrer-policy/#referrer-policy-unsafe-url) is the mapped value to the implementation's ["NEVER_CLEAR"](https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:net/url_request/referrer_policy.h;drc=714e1e34e823b7b4bc172fa4bdefee5aaf5e5fff;l=33)

Reply:
Currently set to the page.
Matt commented under [the other github issue](https://github.com/WICG/turtledove/issues/669#issuecomment-1654050777):
"**Referrer (and initiator) are currently set to the page itself**, though there's a question about changing the initiator, particularly for the bidder requests. We don't follow redirects (largely due to redirect caching potentially being a problem, though reusing bidder scripts is also currently a problem, which we do allow, since caching those between IGs learned on different sites can allow them to share information cached in the JS file itself), **so the referrer policy doesn't currently matter.**"

I can update the referrer and referrer policy to reflect our implementation, and possibly add a note about "no redirect and referrer policy does not currently matter". WDYT?

Reply:
Regarding redirects, the spec already uses the "error" redirect mode, right? So maybe that's fine.

Can you explain the rationale for saying the "referrer policy doesn't currently matter"? I don't understand that.

Reply:
Referrer policy is not useful for the initial request, but for redirects only, if I'm not understanding it wrong? And our requests do not follow redirects.

Also I checked our implementation and there's no referrer being set (confirmed with Matt). Matt said "Seems like we should probably grab the referrer and add it, for consistency, at least for the URLs from the page (Seller URLs)". But before that happens, do I spec referrer as `no-referrer` (then referrer policy really does not matter) for now? I can add a note about some requests' referrer would be set.
As Matt commented in the other github issue, there's also a plan of thinking about changing the request's initiator, which would affect some other fields as well, but that's still not happening.

Reply:
after reading more about referrer policy, seems it does affect the initial request as well (Matt and I thought it does not), so the original statement (referrer policy does not matter when we assumed referrer was set) seems wrong. But turns out our current implementation does not set a referrer.

Reply:
It sounds like the implementation does not manually set a referrer, but does one still get computed and sent anyways based on the URL? My understanding is that the behavior can vary depending on the implementation path taken, and since there are somewhat unique requests, I'd like to know way one way or another for sure if we send any referrer or not (regardless of if we manually _set_ one). A WPT here would be great if possible.

Reply:
No, I used chrome://net-export/ to get network logs for the [publisher page](https://protected-audience-demo-publisher.web.app/) of [our demo](https://protected-audience-demo.web.app/), and non of the requests (bidder and seller script, report url) have referrer. Didn't find signal requests in the log, but it uses the same code as script requests. Matt confirmed that network stack cannot add referrer for a request if it's not manually set, since it doesn't know what a page is.

Request header for seller script (bidder script and report request have very similar headers)
:method: GET
:authority: protected-audience-demo-ssp.web.app
:scheme: https
:path: /decision-logic.js
accept: application/javascript
sec-fetch-site: cross-site
sec-fetch-mode: no-cors
sec-fetch-dest: empty
user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) 

Created a task in backlog to add WPTs for referrer.